crates/ruma-appservice-api/src/event/push_events.rs

```diff
@@ -39,8 +59,29 @@ pub mod v1 {
 
     impl<'a> Request<'a> {
         /// Creates a new `Request` with the given transaction ID and list of events.
-        pub fn new(txn_id: &'a TransactionId, events: &'a [Raw<AnyTimelineEvent>]) -> Self {
-            Self { txn_id, events }
+        pub fn new(
+            txn_id: &'a TransactionId,
+            events: &'a [Raw<AnyTimelineEvent>],
+            #[cfg(feature = "unstable-msc3202")] device_lists: DeviceLists,
+            #[cfg(feature = "unstable-msc3202")] device_one_time_keys_count: BTreeMap<
+                OwnedUserId,
+                BTreeMap<OwnedDeviceId, BTreeMap<DeviceKeyAlgorithm, UInt>>,
+            >,
+            #[cfg(feature = "unstable-msc3202")] device_unused_fallback_key_types: BTreeMap<
+                OwnedUserId,
+                BTreeMap<OwnedDeviceId, Vec<DeviceKeyAlgorithm>>,
+            >,
```

Comment:
Since these fields have default values, they don't need to be passed here, they can be set separately. It's also good to avoid `cfg`-dependent parameters.

```suggestion
```